src/test/kotlin/com/parmet/buf/gradle/AbstractGenerateTest.kt

```diff
@@ -38,4 +38,33 @@ abstract class AbstractGenerateTest : AbstractBufIntegrationTest() {
             listOf("build", "bufbuild", "generated", "java", "com", "google", "type", "DateTime.java")
         assertThat(Paths.get(projectDir.absolutePath, *generatedPathElements.toTypedArray()).toFile().exists()).isTrue()
     }
+
+    @Test
+    fun `buf generate with buf gen template file override`() {
+        gradleRunner().withArguments(BUILD_TASK_NAME).build()
+    }
+
+    @Test
+    fun `buf generate fails with a nonexistent specified template file`() {
+        val result = gradleRunner().withArguments(BUF_GENERATE_TASK_NAME).buildAndFail()
+        assertThat(result.output).contains("Specified templateFileLocation does not exist.")
+    }
+
+    @Test
+    fun `buf generate fails with no default template file and no override specified`() {
+        val result = gradleRunner().withArguments(BUF_GENERATE_TASK_NAME).buildAndFail()
+        assertThat(result.output).contains("No buf.gen.yaml file found in the root directory.")
+    }
+
+    @Test
+    fun `buf generate fails with both default and specified buf gen template files`() {
+        val result = gradleRunner().withArguments(BUF_GENERATE_TASK_NAME).buildAndFail()
+        assertThat(result.output).contains("Buf gen template file specified in the root directory as well as with templateFileLocation")
+    }
+
+    @Test
+    fun `buf generate fails when a specified template file does not exist but a default one does`() {
```

Comment:
Now that I look at it, I usually like to get these tests that involve non-existence to test that things _would have_ worked if the config was right and then modify the `build.gradle` dynamically during the test to break it. In this case that would be:

- have the `templateFileLocation` commented out in the build.gradle file
- run the build and it should succeed (with proto and java files and all)
- remove the `//`
- run the same build and it should now fail

This protects the test from the scenario where no default template file exists, since if it is accidentally deleted then the error condition being checked will still be triggered but it's not the one we actually wanted. This is done elsewhere and you can use it as an example: https://github.com/andrewparmet/buf-gradle-plugin/blob/75d67193c69da80f064ca6daf648a5855bb23745/src/test/kotlin/com/parmet/buf/gradle/AbstractBreakingTest.kt#L42

Reply:
Update the test to pass initially with the commented-out `templateFileLocation` override and then fail when uncommented.